cartography/intel/aws/ec2/tgw.py

```diff
@@ -0,0 +1,214 @@
+import logging
+
+from .util import get_botocore_config
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_transit_gateways(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    return client.describe_transit_gateways()
+
+
+@timeit
+def get_tgw_attachments(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    paginator = client.get_paginator('describe_transit_gateway_attachments')
+    tgw_attachments = []
+    for page in paginator.paginate():
+        tgw_attachments.extend(page['TransitGatewayAttachments'])
+    return {"TransitGatewayAttachments": tgw_attachments}
+
+
+@timeit
+def get_tgw_vpc_attachments(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    paginator = client.get_paginator('describe_transit_gateway_vpc_attachments')
+    tgw_vpc_attachments = []
+    for page in paginator.paginate():
+        tgw_vpc_attachments.extend(page['TransitGatewayVpcAttachments'])
+    return {"TransitGatewayVpcAttachments": tgw_vpc_attachments}
+
+
+@timeit
+def load_transit_gateways(neo4j_session, data, region, current_aws_account_id, aws_update_tag):
+    ingest_transit_gateway = """
+    MERGE (ownerAccount:AWSAccount {id: {OwnerId}})
+    ON CREATE SET ownerAccount.firstseen = timestamp(), ownerAccount.foreign = true
```

Comment:
[comment] Neat, it seems like you really understand the code 👍 ; I am interested if you've seen any foreign accounts here

Reply:
Ha, I tried 😆 
re: foreign accounts, not in our case since we only use TGW attachments across our accounts, but I added it to support the case of cross-organization transit gateway attachments.

Reply:
I like to see the foreign attribute being used and happy that it's used if understood. The question I have on this is are we potentially adding foreign = true to an account we just haven't sync yet but scheduled to be sync later in the flow?

Should we use the foreign flag during post sync analysis vs during ingestion?

Reply:
I actually introduced the same bug initially in :) https://github.com/lyft/cartography/blob/ae6c790c9a799f381b19945a9d2902b6ac5c3ecd/cartography/intel/aws/ec2/vpc_peering.py#L84 . I think this deserve a post sync analysis vs doing it at different stages.

Reply:
I add new bug https://github.com/lyft/cartography/issues/350 to track the foreign account. Feel free to push your code with it.I'll address it in a generic way and cleanup instances.

Reply:
Thanks @sachafaust. While we don't have any foreign accounts attached through TGW we do have a couple of foreign accounts peered through PCX. Since we are running cartography with `--aws-sync-all-profiles` with all org accounts in the boto config, foreign accounts really are non-org accounts for us, so this code works as expected.